lightning/src/onion_message/offers.rs

```diff
@@ -40,7 +43,7 @@ pub trait OffersMessageHandler {
 	/// The returned [`OffersMessage`], if any, is enqueued to be sent by [`OnionMessenger`].
 	///
 	/// [`OnionMessenger`]: crate::onion_message::messenger::OnionMessenger
-	fn handle_message(&self, message: OffersMessage) -> Option<OffersMessage>;
+	fn handle_message<F: Fn(OffersMessage, BlindedPath)>(&self, message: &ReceivedOnionMessage<F, OffersMessage>);
```

Comment:
Couple things here:
- Take `message` by value as it was. That way handlers can take ownership of the argument and reply asynchronously
- `F` should now be bound by `RespondFunction`.

Likewise for `CustomOnionMessageHandler`.

Reply:
Addressed in 9699624
Thanks!